site/documentation/content/api/device-connection/index.md

```diff
@@ -107,6 +107,178 @@ Implementors of this API may return a *404* status code in order to indicate tha
+
+## Get command-handling protocol adapter instances for device
+
+Clients use this command to *get* information about the adapter instances that can handle command & control messages for the given device.
+
+As part of the request message, the client is supposed to provide the list of gateways that may act on behalf of the given device. The client may get this list via the Device Registration API's [*assert Device Registration*]({{< relref "/api/device-registration#assert-device-registration" >}}) operation.
+
+This API doesn't mandate checks on the validity of the given device and the gateway list in order not to introduce a dependency on the *Device Registration API*.
+
+The command implementation MUST determine the adapter instances by applying the following rules (in the given order):
+1. If an adapter instance is associated with the given device, this adapter instance is returned as the *single* returned list entry.
+1. Otherwise, if there is an adapter instance registered for the last known gateway associated with the given device, this adapter instance is returned as the *single* returned list entry. The last known gateway has to be contained in the given list of gateways for this case.
+1. Otherwise, *all* adapter instances associated with any of the given gateway identifiers are returned.
+
+That means that for a device communicating via a gateway, the result is reduced to a *single element* list if an adapter instance for the device itself or its last known gateway is found. The adapter instance registered for the device itself is given precedence in order to ensure that a gateway having subscribed to commands for that *particular device* is chosen over a gateway that has subscribed to commands for *all* devices of a tenant.
+
+**Message Flow**
+
+
+{{< figure src="get_cmd_handling_adapter_instances.svg" title="Client retrieving the list of command-handling adapter instances for a device" alt="A client sends a request message for retrieving the list of command-handling adapter instances for a device and receives a response containing the information" >}}
+
+**Request Message Format**
+
+The following table provides an overview of the properties a client needs to set on a message to retrieve the command-handling adapter instances for a device in addition to the [Standard Request Properties]({{< relref "#standard-request-properties" >}}).
+
+| Name           | Mandatory | Location                 | AMQP Type | Description |
+| :------------- | :-------: | :----------------------- | :-------- | :---------- |
+| *subject*      | yes       | *properties*             | *string*  | MUST be set to `get-cmd-handling-adapter-instances`. |
+| *gateway_ids*  | yes       | *application-properties* | *string*  | A JSON array containing the IDs of the gateways that may act on behalf of the given device. |
```

Comment:
I still believe that we should rather put complex context into the body instead ...
We do this in other cases as well, e.g. https://www.eclipse.org/hono/docs/api/credentials/#get-credentials

Reply:
Well, I don't have a strong opinion here, I can change it back to using the request body.